### PR TITLE
Added my computer and link image dialogs

### PR DESCRIPTION
**Pull request by dougmartin**

### CODE REVIEW

**Comment by dougmartin** [comment]
@knowuh Please review.

File: src/code/views/image-browser-view.coffee [hunk 1]
```diff
@@ -1,202 +1,24 @@
...
+      linkManager: @props.linkManager
 
     (ModalTabbedDialogFactory {title: (tr "~ADD-NEW-IMAGE.TITLE"), close: @props.close, tabs: [
-      ModalTabbedDialog.Tab {label: (tr "~ADD-NEW-IMAGE.IMAGE-SEARCH-TAB"), component: imageSearch}
-      ModalTabbedDialog.Tab {label: (tr "~ADD-NEW-IMAGE.MY-COMPUTER-TAB"), component: (MyComputer {})}
-      ModalTabbedDialog.Tab {label: (tr "~ADD-NEW-IMAGE.LINK-TAB"), component: (Link {})}
+      ModalTabbedDialog.Tab {label: (tr "~ADD-NEW-IMAGE.IMAGE-SEARCH-TAB"), component: (ImageSearch props)}
+      ModalTabbedDialog.Tab {label: (tr "~ADD-NEW-IMAGE.MY-COMPUTER-TAB"), component: (MyComputer props)}
+      ModalTabbedDialog.Tab {label: (tr "~ADD-NEW-IMAGE.LINK-TAB"), component: (Link props)}
     ]})
```
Comment (knowuh) [inline src/code/views/image-browser-view.coffee]: Great cleanup. Thanks.

File: src/code/views/image-search-dialog-view.coffee [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  componentDidMount: ->
+    image = new Image()
+    image.src = @props.imageInfo.image
+    image.onload = =>
+      @setState loaded: true
+  clicked: ->
+    @props.clicked @props.imageInfo
+  render: ->
+    src = if @state.loaded then @props.imageInfo.image else 'img/bb-chrome/spin.svg'
+    if @props.disabled @props.imageInfo
```
Comment (knowuh) [inline src/code/views/image-search-dialog-view.coffee]: At first I thought this was a typo. I was wondering how  `if @props.disabled @props.imageInfo` would evaluate.  

Can you write this so its clearer? It seems like `@props.disabled` is a function and `@props.imageInfo` is the argument, but it doesn't immediately read that way.

File: src/code/views/link-dialog-view.coffee [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      if @state.selectedImage
+        @renderPreviewImage()
+      else
+        (div {},
+          (DropZone {header: (tr "~IMAGE-BROWSER.DROP_IMAGE_FROM_BROWSER"), dropped: @imageDropped}),
+          (p {}, (tr "~IMAGE-BROWSER.TYPE_OR_PASTE_LINK"))
+          (p {}, (tr "~IMAGE-BROWSER.IMAGE_URL"), (input {ref: 'url', type: 'text'}))
+          (p {}, (input {type: 'submit', onClick: @previewImage, value: (tr "~IMAGE-BROWSER.PREVIEW_IMAGE")}))
+        )
+    )
```
Comment (knowuh) [inline src/code/views/link-dialog-view.coffee]: I thought that it was a dialog view for Links (as in node-connection) – which we are going to need in the future. Perhaps we can rename this. I don't have any good name suggestions at the moment, but maybe "external" could be part of it?

File: src/code/views/my-computer-dialog-view.coffee [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    (div {className: 'my-computer-dialog'},
+      if @state.selectedImage
+        @renderPreviewImage()
+      else
+        (div {},
+          (DropZone {header: (tr "~IMAGE-BROWSER.DROP_IMAGE_FROM_DESKTOP"), dropped: @imageDropped}),
+          (p {}, (tr "~IMAGE-BROWSER.CHOOSE_FILE"))
+          (p {}, (input {ref: 'file', type: 'file', onChange: @previewImage}))
+        )
+    )
```
Comment (knowuh) [inline src/code/views/my-computer-dialog-view.coffee]: Again maybe this could have "image" in the name or something.  (Not as critical as the above)

**Comment by knowuh** [comment]
@dougmartin this looks good. Have a look at my comments. I think we should rename link-dialog-view.coffee  (and maybe its `displayName` too) just to avoid future confusion.

**Comment by knowuh** [comment]
Merged.

**Pull request merged**